nixos/modules/services/monitoring/vnstat.nix

```diff
@@ -0,0 +1,42 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.vnstat;
+in {
+  options.services.vnstat = {
+    enable = mkOption {
+      default = false;
+      description = ''
+        Wether to enable update of network usage statistics via vnstatd.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.extraUsers.vnstatd = {
+      isSystemUser = true;
+      description = "vnstat daemon user";
+      home = "/var/lib/vnstat";
+      createHome = true;
+    };
+
+    systemd.services.vnstat = {
+      description = "vnStat network traffic monitor";
+      path = [ pkgs.coreutils ];
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      unitConfig.documentation = "man:vnstatd(1) man:vnstat(1) man:vnstat.conf(5)";
+      preStart = "chmod 755 /var/lib/vnstat";
```

Comment:
That chmod is there so others have read permissions on _/var/lib/vnstat/_
As long as _vnstatd_ is the owner of _/var/lib/vnstat_ I believe that chmod will succeed. If _/var/lib/vnstat_ is owned by someone else, a chmod alone is not sufficient to fix this. Should I add also a chown into the preStart rule? For me that would feel a little weird.